hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkDataSet.java

```diff
@@ -789,43 +773,17 @@ public static StructType generateTableSchema(OperationContext context) throws St
                                           String compression,
                                           OperationContext context) throws StandardException
     {
-        compression = SparkDataSet.getAvroCompression(compression);
 
-        StructType dataSchema = null;
         StructType tableSchema = generateTableSchema(context);
-
-        // what is this? why is this so different from parquet/orc ?
-        // actually very close to NativeSparkDataSet.writeFile
-        dataSchema = ExternalTableUtils.getDataSchema(dsp, tableSchema, partitionBy, location, "a");
-
+        StructType dataSchema = SparkExternalTableUtil.getDataSchemaAvro(dsp, tableSchema, partitionBy, location);
         if (dataSchema == null)
             dataSchema = tableSchema;
-
         Dataset<Row> insertDF = SpliceSpark.getSession().createDataFrame(
-                rdd.map(new SparkSpliceFunctionWrapper<>(new CountWriteFunction(context))).map(new LocatedRowToRowAvroFunction()),
+                rdd.map(new SparkSpliceFunctionWrapper<>(new EmptyFunction())).map(new LocatedRowToRowAvroFunction()),
```

Comment:
Do we need the EmptyFunction here?

Reply:
i'm not exactly sure how to remove it

Reply:
if i remove it, it doesn't compile anymore